tests/main/prepare-image-classic/task.yaml

```
@@ -0,0 +1,72 @@
+summary: Check that prepare-image can prepare UC20+ classic models
+
+# running this on one machine is enough
+systems: [ubuntu-22.04-64]
+
+environment:
+    ROOT: /home/test/tmp/
+    STORE_ADDR: localhost:11028
+    STORE_DIR: $(pwd)/fake-store-blobdir
+
+prepare: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+    #shellcheck source=tests/lib/store.sh
+    . "$TESTSLIB"/store.sh
+    setup_fake_store "$STORE_DIR"
+
+    mkdir "$ROOT"
+    chown -R test:test "$ROOT"
+
+restore: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+    #shellcheck source=tests/lib/store.sh
+    . "$TESTSLIB"/store.sh
+    teardown_fake_store "$STORE_DIR"
+    rm -rf "$ROOT"
+
+execute: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+    if [ "$REMOTE_STORE" = "staging" ]; then
+        echo "SKIP: cannot run with staging store yet"
+        exit 0
+    fi
+
+    # get the model assertion
+    gendeveloper1 sign-model < "$TESTSLIB"/assertions/developer1-22-classic-dangerous.json > "$ROOT/model.assertion"
+    
+    echo Expose the needed assertions through the fakestore
+    cp "$TESTSLIB"/assertions/developer1.account "$STORE_DIR/asserts"
+    cp "$TESTSLIB"/assertions/developer1.account-key "$STORE_DIR/asserts"
+    cp "$TESTSLIB"/assertions/testrootorg-store.account-key "$STORE_DIR/asserts"
+    # have snap use the fakestore for assertions (but nothing else)
+    export SNAPPY_FORCE_SAS_URL=http://$STORE_ADDR
+
+    echo Running prepare-image as a user
+    su -c "SNAPPY_USE_STAGING_STORE=$SNAPPY_USE_STAGING_STORE snap prepare-image --classic --channel edge $ROOT/model.assertion $ROOT" test 2>stderr
+
+    # prepare-image ran as user so it warns about the ownership
+    MATCH 'WARNING: ensure that the contents under .* are owned by root:root in the \(final\) image' < stderr
```

It's not clear why this is here. It does not seem to be relevant for this test. I'd just remove it, or, if the goal is to ensure that this is the only line in the stderr, I would also do this (untested):

```suggestion
    # Ensure that only the expected errors have occurred:
    wc -l stderr | MATCH "^1$"
    # prepare-image ran as user so it warns about the ownership
    MATCH 'WARNING: ensure that the contents under .* are owned by root:root in the \(final\) image' < stderr
```